moveit_simple/src/moveit_simple.cpp

```diff
@@ -219,16 +217,115 @@ bool Robot::getJointSolution(const Eigen::Affine3d &pose, double timeout,
   return getIK(pose, local_seed, joint_point, timeout);
 }
 
+bool Robot::getJointSolution(const Eigen::Affine3d &pose, const std::string& custom_tool_frame,
+                             double timeout, const std::vector<double> & seed,
+                             std::vector<double> & joint_point) const
+{
+
+  // Ideally "tool0" should be replaced with joint_group_->getEndEffectorName()
+  std::string moveit_tool_link = "tool0";
+
+  // Transform Target/Goal Point from custom tool frame to moveit_end_link
+  const Eigen::Affine3d custom_frame_goal_pose = 
+                        transformPoseBetweenFrames(pose, custom_tool_frame, 
+                                                   moveit_tool_link);
+
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  std::vector<double> local_seed = seed;
+  if (seed.empty())
+  {
+    ROS_INFO_STREAM("Empty seed passed to getJointSolution, using current state");
+    local_seed =  getJointState();
+  }
+
+  return getIK(custom_frame_goal_pose, local_seed, joint_point, timeout);
+}
+
+
+Eigen::Affine3d Robot::transformPoseBetweenFrames(const Eigen::Affine3d &target_pose,
+                                              const std::string& frame_in,
+                                              const std::string& frame_out) const
+{
+
+  if(frame_in.compare(frame_out) == 0 && 
+    robot_model_ptr_->hasLinkModel(frame_in) &&
+    robot_model_ptr_->hasLinkModel(frame_out)) {
+
+      ROS_INFO_STREAM("Returning same target_pose");
+
+      return target_pose;
+  }
+
+  //if( tf_buffer_.canTransform(joint_group_->getEndEffectorName(), ros::Time::now(), ros::Duration(0.1)) ) {
+    try
+    {
+        ROS_INFO_STREAM("Looked up tf named frame: " << frame_in);
```

Comment:
Add more context here as well....looking up frame because it doesn't exist in Robot model.